libraries/chain/block_log.cpp

```diff
@@ -615,6 +628,68 @@ namespace eosio { namespace chain {
             FC_LOG_AND_RETHROW()
          }
 
+         std::vector<char> read_serialized_block_by_num(uint32_t block_num, verify_block_num_t verify_block_num) final {
+            try {
+               uint64_t pos = get_block_pos(block_num);
+
+               if (pos == block_log::npos || !head) {
+                  // Rare case. No need a special code for it.
+                  // Just fall back to the regular `retry_read_block_by_num` and then serialize.
+                  auto ret = retry_read_block_by_num(block_num);
+                  if (ret) {
+                     return fc::raw::pack(*ret);
+                  } else {
+                     return {};
+                  }
+               }
+
+               if (verify_block_num == verify_block_num_t::yes) {
```

Comment:
Or should we always do a full deserialization here to make sure the block is valid but still return the serialized format? This matches the validation behavior of the original `read_block_by_num()` but still saves Net Plugin from doing serialization.

Reply:
Wouldn't a block in the block log be guaranteed to be valid?

Reply:
I don't understand whay we need this `verify_block_num`. I would think that the blocks in the block log are correct, and that therefore the block at position ` get_block_pos(block_num)` is guaranteed to have the block number `block_num`. Is that not the case?

Reply:
The original `read_block_by_num` checks `block_num`. I am little paranoid. Will discuss in design meeting.